src/commands/schema_diff.ts

```diff
@@ -0,0 +1,123 @@
+import { BranchScopeProps } from '../types';
+import { createPatch } from 'diff';
+import { Database } from '@neondatabase/api-client';
+import chalk from 'chalk';
+import { writer } from '../writer.js';
+import { branchIdFromProps, branchIdResolve } from '../utils/enrichers.js';
+
+type SchemaDiffProps = BranchScopeProps & {
+  branch?: string;
+  baseBranch?: string;
+  compareBranch: string;
+  database: string;
+};
+
+const COLORS = {
+  added: chalk.green,
+  removed: chalk.red,
+  header: chalk.yellow,
+  section: chalk.magenta,
+};
+
+type ColorId = keyof typeof COLORS;
+
+export const schemaDiff = async (props: SchemaDiffProps) => {
+  props.branch = props.baseBranch || props.branch;
+  const baseBranch = await branchIdFromProps(props);
+  const compareBranch = await branchIdResolve({
+    branch: props.compareBranch,
+    apiClient: props.apiClient,
+    projectId: props.projectId,
+  });
+
+  if (baseBranch === compareBranch) {
+    throw new Error(
+      'Can not compare the branch with itself. Please specify different branch to compare.',
+    );
+  }
+
+  const baseDatabases = await fetchDatabases(baseBranch, props);
+
+  if (props.database) {
+    const database = baseDatabases.find((db) => db.name === props.database);
+
+    if (!database) {
+      throw new Error(
+        `Database ${props.database} does not exist in base branch ${baseBranch}`,
+      );
+    }
+
+    const patch = await createSchemaDiff(
+      baseBranch,
+      compareBranch,
+      database,
+      props,
+    );
+    writer(props).text(colorize(patch));
+    return;
+  }
+
+  baseDatabases.map(async (database) => {
+    const patch = await createSchemaDiff(
+      baseBranch,
+      compareBranch,
+      database,
+      props,
+    );
+    writer(props).text(colorize(patch));
```

Comment:
not sure if this is important, but the order of output here is random

Reply:
yes, the order is not important